### PR TITLE
21 implement chunk array operation

### PR DESCRIPTION
**Pull request by JenDevs**
@coderabbitai

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality to segment arrays and lists into smaller, manageable groups with robust error handling for invalid or unexpected inputs.

- **Tests**
  - Expanded automated testing to cover diverse scenarios including edge cases and performance, ensuring the new chunking features reliably handle various input conditions.

### CODE REVIEW

File: src/main/java/org/fungover/breeze/util/Arrays.java [hunk 1]
```diff
@@ -71,4 +75,108 @@ public static <T> T[][] transpose(T[][] array) {
...
+            for(T element : array) {
+                if (element == null) {
+                    throw new IllegalArgumentException("Element must not be null");
+                }
+            }
+        }
+
+        if (size <= 0) {
+            throw new IllegalArgumentException("Size must be greater than 0");
+        } else if (array.length == 0) {
```
Comment (Biggienation) [inline src/main/java/org/fungover/breeze/util/Arrays.java:128]: Im not sure I get what this "else if"  block is doing.  Is there any reason that we are returning a empty 2D array?

Comment (JenDevs) [reply src/main/java/org/fungover/breeze/util/Arrays.java:128]: Good point! I refactored the if-statements into if-else and must have included this one by mistake. It should have its own if-statement. It returns an empty 2D array to handle empty arrays, as specified in the issue.

File: src/main/java/org/fungover/breeze/util/Arrays.java [hunk 2]
```diff
@@ -71,4 +75,110 @@ public static <T> T[][] transpose(T[][] array) {
...
+     */
+    public static <T>List<List<T>> chunkList(List<T> list, int size) {
+        if(list == null) {
+            throw new IllegalArgumentException("Input list must not be null");
+        } else {
+            for(T element : list) {
+                if (element == null) {
+                    throw new IllegalArgumentException("Element must not be null");
+                }
+            }
```
Comment (ehottinen) [inline src/main/java/org/fungover/breeze/util/Arrays.java:170]: I don't see why null values in the lists would cause issues, what's the reason for throwing an exception here?

File: src/main/java/org/fungover/breeze/util/Arrays.java [hunk 3]
```diff
@@ -71,4 +75,110 @@ public static <T> T[][] transpose(T[][] array) {
...
+
+        List<List<T>> chunks = new ArrayList<>();
+        for (int i = 0; i < list.size(); i+= size) {
+            chunks.add(List.copyOf(list.subList(i, Math.min(list.size(), i + size))));
+        }
+
+        return Collections.unmodifiableList(chunks);
+    }
+
 }
```
Comment (ehottinen) [inline src/main/java/org/fungover/breeze/util/Arrays.java:184]: The code is easy to read and follows a consistent structure. Nicely done.

File: src/test/java/org/fungover/breeze/util/ArraysTest.java [hunk 4]
```diff
@@ -123,4 +127,254 @@ void preservedTypeInformation() {
...
+
+        long startTime = System.nanoTime();
+        List<List<Integer>> result = Arrays.chunkList(benchmarkList, size);
+        long endTime = System.nanoTime();
+        long elapsedTime = (endTime - startTime) / 1_000_000;
+
+        assertThat(result).hasSize(benchmarkList.size() / size);
+        System.out.println("Large list test complete in " + elapsedTime + " ms");
+    }
+
```
Comment (ehottinen) [inline src/test/java/org/fungover/breeze/util/ArraysTest.java:378]: You have written very thorough and well-structured tests. They cover a wide range of edge cases and scenarios, making the code highly robust. Great job!

**Review by ehottinen — APPROVED** [review]

**Review by Biggienation — APPROVED** [review]
Easy to read code and code coverage is good. Nicely done
